packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/ConnectionState.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.mongodb.sync
+
+/**
+ * A **connection state** indicates the state of the underlying connection of the [SyncSession].
+ */
+public enum class ConnectionState {
```

Comment:
Having this as a top-level class makes sense IMO.  Maybe a stretch, but my line of thinking is that that `SyncSession.State` is directly related to the `SyncSession`, similar to `User.State` whereas, ConnectionState is a little more outside the responsibility of SyncSession.